Bug:The library does not show all the tracks present in the folder.

### What happened?

The library does not show all the tracks present in the folder.


### Steps to Reproduce

Try to rescan = no change

Exemple:

<img width="1423" height="552" alt="Image" src="https://github.com/user-attachments/assets/4c9a331e-0bc2-4a52-84af-006bcf0fe008" />

### Expected Behavior

10 track in the folder, 10 tracks in the library...

### Is the app crashing?

No - Just unexpected behavior

### Petrichor Version

1.3.0

### macOS Version

macOS 26 (Tahoe)

### Logs

```shell
A lot of this in the log...
[2025-12-10 19:31:15.988] [ERROR] [DMTrackProcessing > processBatch:63] Failed to add new track Dadance: SQLite error 19: CHECK constraint failed: ("rating" IS NULL) OR (("rating" >= 0) AND ("rating" <= 5)) - while executing `INSERT INTO "tracks" ("id", "folder_id", "path", "filename", "title", "artist", "album", "composer", "genre", "year", "duration", "format", "date_added", "track_artwork_data", "is_favorite", "play_count", "last_played_date", "album_artist", "track_number", "total_tracks", "disc_number", "total_discs", "rating", "compilation", "release_date", "original_release_date", "bpm", "media_type", "bitrate", "sample_rate", "channels", "codec", "bit_depth", "lossless", "file_size", "date_modified", "is_duplicate", "primary_track_id", "duplicate_group_id", "sort_title", "sort_artist", "sort_album", "sort_album_artist", "album_id", "extended_metadata") VALUES (?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?,?)`
```

### Before submitting

- [x] I searched existing issues to avoid duplicates
- [x] I'm running macOS 14.0 or higher

--- a/Core/MetadataExtractor.swift
+++ b/Core/MetadataExtractor.swift
@@ -604,16 +604,25 @@ class MetadataExtractor {
     private static func extractRating(from rawRating: Int?) -> Int? {
         guard let raw = rawRating, raw > 0 else { return nil }
         
-        // Default rating range
+        let normalized: Int
+        
+        // Default rating range (1-5)
         if raw <= 5 {
-            return raw
+            normalized = raw
+        }
+        // ID3v2 POPM rating range (1-255 mapped to 1-5)
+        else if raw <= 31 {
+            normalized = 1
+        } else if raw <= 95 {
+            normalized = 2
+        } else if raw <= 159 {
+            normalized = 3
+        } else if raw <= 223 {
+            normalized = 4
+        } else {
+            normalized = 5
         }
         
-        // ID3v2 POPM rating range
-        if raw <= 31 { return 1 }
-        if raw <= 95 { return 2 }
-        if raw <= 159 { return 3 }
-        if raw <= 223 { return 4 }
-        return 5
+        return min(max(normalized, 0), 5)
     }
 }

